pkg/codegen/java/gen_defaults.go

```diff
@@ -73,10 +79,10 @@ func (dg *defaultsGen) builderExpr(
 		} else {
 			builderTransformCode = ".output()"
 		}
-	} else {
+	} else if prop.Secret {
 		// TODO this should be handled or an error
-		fmt.Printf("WARN: secret propety does not have an Output type: %v\n",
-			prop.Name)
+		fmt.Printf("WARN: secret propety %s (%s) does not have an Output type\n",
+			prop.Name, propContext)
 	}
```

Comment:
Is it safe to ignore the `else` here? The code before changes warned about all unrecognized cases.

If it is safe to ignore, maybe adding an explicit ignore would be a nice improvement for readability:
```
} else {
  // ... description of why we ignore this branch ...
}
```

Also we might consider throwing an error instead. 

I'm not sure what is the best course of action here, but just wanted to point out that after the change we silently ignore more then before the change.

Reply:
I think so. 

All the other warnings were false warnings. If a Config property for example has type String but is not marked secret (very common), this is all good, we should not warn. 

I like the the idea of an no-op else, let me try that. 

We cannot throw on this since this would cost us the pulumi-azure-native provider. It needs this until there is a deeper fix. So it has to be a warning for now.

Reply:
>  after the change we silently ignore more then before the change.

Intentionally, yes! Those were spurious warnings. It was warning on AWS for example where all is well.